### PR TITLE
Struct.new passes same subclass to the block

### PR DESCRIPTION
**Pull request by kachick**
Follow #151 

```shell
$ ../mspec/bin/mspec core/struct
ruby 2.3.0dev (2015-10-18 trunk 52184) [x86_64-darwin14]
................................................................................................

Finished in 0.018951 seconds

28 files, 85 examples, 147 expectations, 0 failures, 0 errors
```

### CODE REVIEW

File: core/struct/new_spec.rb [hunk 1]
```diff
@@ -71,26 +71,49 @@ def obj.to_sym() :ruby end
...
+
+      instance.version.should == '2.2'
+      instance.platform.should == :ruby
+    end
+
+    it "passes same struct class to the block" do
+      given = nil
+      Struct.new(:attr) do |block_parameter|
+        given = block_parameter
+      end.should equal(given)
```
Comment (eregon) [inline core/struct/new_spec.rb]: This just tests the block return value.
It should be
```ruby
given = nil
klass = Struct.new(:attr) do |block_parameter|
  given = block_parameter
end
given.should equal(klass)
```

Comment (eregon) [inline core/struct/new_spec.rb]: My fault, the return value is indeed always the new struct class.
But I think it's still easier to read this way.

Comment (kachick) [inline core/struct/new_spec.rb]: >But I think it's still easier to read this way.

:+1:  I agree with it :smile: 
I'll update this patch!

**Comment by eregon** [comment]
Looks good, thanks!
Please merge yourself  when you updated about little nit :smiley:

**Comment by kachick** [comment]
Thanks for your review!

**Pull request merged**